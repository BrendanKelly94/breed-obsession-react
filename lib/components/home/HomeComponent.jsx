import React from 'react';
import SelectComponent from './SelectComponent.jsx';
import { Link } from 'react-router-dom';


const HomeComponent = ({
  breeds,
  cities,
  setBreed,
  setCity,
  setSex,
  setSize,
  setAge,
  fetchCities,
  fetchBreeds,
  requestingBreeds,
  requestingCities,
}) => {

  const animals = ['Barnyard', 'Bird', 'Cat', 'Dog' , 'Horse', 'Reptile', 'Smallfurry'];

  const states = [
    'Alabama','Alaska','American Samoa','Arizona','Arkansas','California','Colorado',
    'Connecticut','Delaware','District of Columbia','Federated States of Micronesia','Florida','Georgia',
    'Guam','Hawaii','Idaho','Illinois','Indiana','Iowa','Kansas','Kentucky','Louisiana','Maine',
    'Marshall Islands','Maryland','Massachusetts','Michigan','Minnesota','Mississippi','Missouri','Montana',
    'Nebraska','Nevada','New Hampshire','New Jersey','New Mexico','New York','North Carolina','North Dakota',
    'Northern Mariana Islands','Ohio','Oklahoma','Oregon','Palau','Pennsylvania','Puerto Rico','Rhode Island',
    'South Carolina','South Dakota','Tennessee','Texas','Utah','Vermont','Virgin Island','Virginia','Washington',
    'West Virginia','Wisconsin','Wyoming'
  ];

  return (
    <div id = "home-component">
      <div className = "row" id = "home-name">
        <h1> Breed Obsession </h1>
      </div>

      <div className = "row" id = "home">
        <SelectComponent isRequesting = {false} label = "Animal" items = {animals} callback = {fetchBreeds} />
        <SelectComponent isRequesting = {requestingBreeds} label = "Breed" items = {breeds} callback = {setBreed} />
        <SelectComponent isRequesting = {false} label = "State" items = {states} callback = {fetchCities} />
        <SelectComponent isRequesting = {requestingCities} label = "City" items = {cities} callback = {setCity} />

        <div className = "row"  id = "advanced-toggle">
          <button data-toggle = "collapse" data-target = "#advanced" aria-expanded = "false" aria-controls = "advanced">
            Advanced Search
          </button>
        </div>

        <div className = "collapse" id = "advanced">
          <SelectComponent isRequesting = {false} label = "Sex" items = {['M', 'F']} callback = {setSex} />
          <SelectComponent isRequesting = {false} label = "Age" items = {['Baby', 'Young Adult', 'Senior']} callback = {setAge} />
          <SelectComponent isRequesting = {false} label = "Size" items = {['S', 'M', 'L', 'XL']} callback = {setSize} />
        </div>

      </div>

      <div className = "row" id = "search">
        <Link to = "/postings">
          <button className = "btn btn-primary"> Go </button>
        </Link>
      </div>

      <footer>
        <div className = "container">
          <p>Powered by Petfinder <a href = "https:www.petfinder.com">www.petfinder.com</a></p>
        </div>
      </footer>
    </div>
  );
};

export default HomeComponent;
