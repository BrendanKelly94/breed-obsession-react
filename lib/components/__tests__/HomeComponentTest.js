import React from 'react';
import HomeComponent from '../home/HomeComponent.jsx';
import { MemoryRouter } from 'react-router-dom';

import renderer from 'react-test-renderer';

describe('HomeComponent', () => {

  const testProps = {
    breeds: [],
    cities: [],
    setBreed: jest.fn(() => ({})),
    setCity: jest.fn(() => ({})),
    setSex: jest.fn(() => ({})),
    setSize: jest.fn(() => ({})),
    setAge: jest.fn(() => ({})),
    fetchCities: jest.fn(() => ({})),
    fetchBreeds: jest.fn(() => ({}))
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <HomeComponent
          {...testProps}
        />
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
