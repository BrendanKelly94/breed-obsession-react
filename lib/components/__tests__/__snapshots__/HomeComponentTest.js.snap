// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent renders correctly 1`] = `
<div
  id="home-component"
>
  <div
    className="row"
    id="home-name"
  >
    <h1>
       Breed Obsession 
    </h1>
  </div>
  <div
    className="row"
    id="home"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          Animal
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        >
          <option
            value="barnyard"
          >
             
            barnyard
             
          </option>
          <option
            value="bird"
          >
             
            bird
             
          </option>
          <option
            value="cat"
          >
             
            cat
             
          </option>
          <option
            value="dog"
          >
             
            dog
             
          </option>
          <option
            value="horse"
          >
             
            horse
             
          </option>
          <option
            value="reptile"
          >
             
            reptile
             
          </option>
          <option
            value="smallfurry"
          >
             
            smallfurry
             
          </option>
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          Breed
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          State
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        >
          <option
            value="Alabama"
          >
             
            Alabama
             
          </option>
          <option
            value="Alaska"
          >
             
            Alaska
             
          </option>
          <option
            value="American Samoa"
          >
             
            American Samoa
             
          </option>
          <option
            value="Arizona"
          >
             
            Arizona
             
          </option>
          <option
            value="Arkansas"
          >
             
            Arkansas
             
          </option>
          <option
            value="California"
          >
             
            California
             
          </option>
          <option
            value="Colorado"
          >
             
            Colorado
             
          </option>
          <option
            value="Connecticut"
          >
             
            Connecticut
             
          </option>
          <option
            value="Delaware"
          >
             
            Delaware
             
          </option>
          <option
            value="District of Columbia"
          >
             
            District of Columbia
             
          </option>
          <option
            value="Federated States of Micronesia"
          >
             
            Federated States of Micronesia
             
          </option>
          <option
            value="Florida"
          >
             
            Florida
             
          </option>
          <option
            value="Georgia"
          >
             
            Georgia
             
          </option>
          <option
            value="Guam"
          >
             
            Guam
             
          </option>
          <option
            value="Hawaii"
          >
             
            Hawaii
             
          </option>
          <option
            value="Idaho"
          >
             
            Idaho
             
          </option>
          <option
            value="Illinois"
          >
             
            Illinois
             
          </option>
          <option
            value="Indiana"
          >
             
            Indiana
             
          </option>
          <option
            value="Iowa"
          >
             
            Iowa
             
          </option>
          <option
            value="Kansas"
          >
             
            Kansas
             
          </option>
          <option
            value="Kentucky"
          >
             
            Kentucky
             
          </option>
          <option
            value="Louisiana"
          >
             
            Louisiana
             
          </option>
          <option
            value="Maine"
          >
             
            Maine
             
          </option>
          <option
            value="Marshall Islands"
          >
             
            Marshall Islands
             
          </option>
          <option
            value="Maryland"
          >
             
            Maryland
             
          </option>
          <option
            value="Massachusetts"
          >
             
            Massachusetts
             
          </option>
          <option
            value="Michigan"
          >
             
            Michigan
             
          </option>
          <option
            value="Minnesota"
          >
             
            Minnesota
             
          </option>
          <option
            value="Mississippi"
          >
             
            Mississippi
             
          </option>
          <option
            value="Missouri"
          >
             
            Missouri
             
          </option>
          <option
            value="Montana"
          >
             
            Montana
             
          </option>
          <option
            value="Nebraska"
          >
             
            Nebraska
             
          </option>
          <option
            value="Nevada"
          >
             
            Nevada
             
          </option>
          <option
            value="New Hampshire"
          >
             
            New Hampshire
             
          </option>
          <option
            value="New Jersey"
          >
             
            New Jersey
             
          </option>
          <option
            value="New Mexico"
          >
             
            New Mexico
             
          </option>
          <option
            value="New York"
          >
             
            New York
             
          </option>
          <option
            value="North Carolina"
          >
             
            North Carolina
             
          </option>
          <option
            value="North Dakota"
          >
             
            North Dakota
             
          </option>
          <option
            value="Northern Mariana Islands"
          >
             
            Northern Mariana Islands
             
          </option>
          <option
            value="Ohio"
          >
             
            Ohio
             
          </option>
          <option
            value="Oklahoma"
          >
             
            Oklahoma
             
          </option>
          <option
            value="Oregon"
          >
             
            Oregon
             
          </option>
          <option
            value="Palau"
          >
             
            Palau
             
          </option>
          <option
            value="Pennsylvania"
          >
             
            Pennsylvania
             
          </option>
          <option
            value="Puerto Rico"
          >
             
            Puerto Rico
             
          </option>
          <option
            value="Rhode Island"
          >
             
            Rhode Island
             
          </option>
          <option
            value="South Carolina"
          >
             
            South Carolina
             
          </option>
          <option
            value="South Dakota"
          >
             
            South Dakota
             
          </option>
          <option
            value="Tennessee"
          >
             
            Tennessee
             
          </option>
          <option
            value="Texas"
          >
             
            Texas
             
          </option>
          <option
            value="Utah"
          >
             
            Utah
             
          </option>
          <option
            value="Vermont"
          >
             
            Vermont
             
          </option>
          <option
            value="Virgin Island"
          >
             
            Virgin Island
             
          </option>
          <option
            value="Virginia"
          >
             
            Virginia
             
          </option>
          <option
            value="Washington"
          >
             
            Washington
             
          </option>
          <option
            value="West Virginia"
          >
             
            West Virginia
             
          </option>
          <option
            value="Wisconsin"
          >
             
            Wisconsin
             
          </option>
          <option
            value="Wyoming"
          >
             
            Wyoming
             
          </option>
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          City
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          Sex
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        >
          <option
            value="M"
          >
             
            M
             
          </option>
          <option
            value="F"
          >
             
            F
             
          </option>
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          Age
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        >
          <option
            value="Baby"
          >
             
            Baby
             
          </option>
          <option
            value="Young Adult"
          >
             
            Young Adult
             
          </option>
          <option
            value="Senior"
          >
             
            Senior
             
          </option>
        </select>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-md-4 col-lg-4"
      >
        <label>
          Size
        </label>
      </div>
      <div
        className="col-sm-8 col-md-8 col-lg-8"
      >
        <select
          onChange={[MockFunction]}
        >
          <option
            value="S"
          >
             
            S
             
          </option>
          <option
            value="M"
          >
             
            M
             
          </option>
          <option
            value="L"
          >
             
            L
             
          </option>
          <option
            value="XL"
          >
             
            XL
             
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="row"
    id="search"
  >
    <a
      href="/postings"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
      >
         Go 
      </button>
    </a>
  </div>
  <footer>
    <div
      className="container"
    >
      <p>
        Powered by Petfinder 
        <a
          href="https:www.petfinder.com"
        >
          www.petfinder.com
        </a>
      </p>
    </div>
  </footer>
</div>
`;
